sled-agent/src/services.rs

```diff
@@ -463,12 +463,26 @@ impl ServiceManager {
         Ok(None)
     }
 
+    // Check the services intended to run in the zone to determine whether any
+    // additional privileges need to be enabled for the zone.
+    fn privs_needed(&self, req: &ServiceZoneRequest) -> Vec<String> {
+        let mut needed = Vec::new();
+        for svc in &req.services {
+            if let ServiceType::Tfport { .. } = svc {
+                needed.push("default".to_string());
+                needed.push("sys_dl_config".to_string());
+            }
+        }
```

Comment:
Nit: This is a privilege which is being set for the entire zone, not just this service, right?

Although it is technically *used* for the `Tfport` service, do you think it would be more clear to match on `ZoneType::Switch`, rather than searching for a zone containing the `Tfport` service?

Reply:
It's only needed for the `tfport` service.  If you launch a stub switch zone without that service, the additional privilege isn't necessary.

Reply:
Fair enough!